src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/Documents/VisualStudioEditorDocumentManager.cs

```diff
@@ -52,8 +53,12 @@ internal class VisualStudioEditorDocumentManager : EditorDocumentManagerBase
             _runningDocumentTable = (IVsRunningDocumentTable4)runningDocumentTable;
             _editorAdaptersFactory = editorAdaptersFactory;
 
-            var hr = runningDocumentTable.AdviseRunningDocTableEvents(new RunningDocumentTableEventSink(this), out _);
-            Marshal.ThrowExceptionForHR(hr);
+            joinableTaskContext.Factory.Run(() =>
```

Comment:
I think a comment here that `AdviseRunningDocTableEvents` is required to run on the UI thread might be helpful for future readers